app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt

```diff
@@ -144,26 +144,31 @@ class TabTrayFragment : Fragment(R.layout.fragment_tab_tray), TabsTray.Observer,
         val tabs = getListOfSessions()
         val tabIds = tabs.map { it.id }.toList().toTypedArray()
         val tabCollectionStorage = (activity as HomeActivity).components.core.tabCollectionStorage
+        val navController = findNavController()
 
         val step = when {
+            // Show the SelectTabs fragment if there are multiple opened tabs to select which tabs
+            // you want to save to a collection.
+            tabs.size > 1 -> SaveCollectionStep.SelectTabs
             // If there is an existing tab collection, show the SelectCollection fragment to save
             // the selected tab to a collection of your choice.
             tabCollectionStorage.cachedTabCollections.isNotEmpty() -> SaveCollectionStep.SelectCollection
             // Show the NameCollection fragment to create a new collection for the selected tab.
             else -> SaveCollectionStep.NameCollection
         }
 
+        if (navController.currentDestination?.id == R.id.collectionCreationFragment) return
+
+        // Only register the observer right before moving to collection creation
+        // registerCollectionStorageObserver()
```

Comment:
@boek I left this in the patch but commented out, as it's in the HomeFragment's implementation, but I don't understand what it's doing for us.  Do we need this?

Also, upon saving to collection the tabs don't go away.  Is that something we need to be doing?

Reply:
We don't have collections on the tab tray so we don't need to register for that observer here!

Reply:
And no, I think that behavior was confusing so we stopped removing the tabs that we were saving to the collection